### ISSUE TITLE
Helper methods for working with SpatiaLite

### ISSUE THREAD

**Issue by simonw** [simonw/sqlite-utils#79]
As demonstrated by this piece of documentation, using SpatiaLite with sqlite-utils requires a fair bit of boilerplate:
https://github.com/simonw/sqlite-utils/blob/f7289174e66ae4d91d57de94bbd9d09fabf7aff4/docs/python-api.rst#L880-L909

**Comment by eyeseast** [comment]
Came here to add this. I might pick it up.

Would also add a utility to create (and update and delete?) a spatial index. It's not much code but I have to look it up every time.

**Comment by eyeseast** [comment]
Some proposals I'd add to sqlite-utils:

Some version of this, from [geojson-to-sqlite](https://github.com/simonw/geojson-to-sqlite/blob/main/geojson_to_sqlite/utils.py#L124-L130):

```python
def init_spatialite(db, lib):
    db.conn.enable_load_extension(True)
    db.conn.load_extension(lib)
    # Initialize SpatiaLite if not yet initialized
    if "spatial_ref_sys" in db.table_names():
        return
    db.conn.execute("select InitSpatialMetadata(1)")
```

Also a function for creating a spatial index:

```python
db.conn.execute("select CreateSpatialIndex(?, ?)", [table, "geometry"])
```

I don't know the nuances of updating a spatial index, or checking if one already exists. This could be a CLI method like:

```sh
sqlite-utils spatial-index spatial.db table-name column-name
```

**Comment by eyeseast** [comment]
One more thing I'm going to add: A method to add a geometry column, which I'll need to do to create a spatial index on a table.

**Comment by eyeseast** [comment]
@simonw I have a PR here https://github.com/simonw/sqlite-utils/pull/385 that adds Spatialite helpers on the Python side. Please let me know how it looks.

**Comment by eyeseast** [comment]
Taking part of the conversation from #385 here.

> Would sqlite-utils add-geometry-column ... be a good CLI enhancement. for example?

Yes. And also `sqlite-utils create-spatial-index` would be great to have. My plan would be to add those once the Python API is settled.

**Comment by simonw** [comment]
Documentation: https://sqlite-utils.datasette.io/en/latest/python-api.html#spatialite-helpers

**Comment by simonw** [comment]
Shipped in 3.23: https://sqlite-utils.datasette.io/en/stable/changelog.html#v3-23
